app/src/main/kotlin/com/simplemobiletools/gallery/activities/ViewPagerActivity.kt

```diff
@@ -454,52 +455,60 @@ class ViewPagerActivity : SimpleActivity(), ViewPager.OnPageChangeListener, View
         SaveAsDialog(this, currPath, false) {
             Thread({
                 toast(R.string.saving)
-                if (it.isJpg() && !isPathOnSD(it)) {
-                    if (it == currPath) {
-                        rotateFileByExif(it)
-                        runOnUiThread {
-                            (getCurrentFragment() as? PhotoFragment)?.refreshBitmap()
+                val selectedFile = File(it)
+                val tmpFile = File(filesDir, ".tmp_${it.getFilenameFromPath()}")
+                try {
+                    val bitmap = BitmapFactory.decodeFile(currPath)
+                    getFileOutputStream(tmpFile) {
+                        if (it == null) {
+                            toast(R.string.unknown_error_occurred)
+                            deleteFile(tmpFile) {}
+                            return@getFileOutputStream
                         }
-                    } else {
-                        copyFile(currPath, it)
-                        rotateFileByExif(it)
+
+                        if (currPath.isJpg()) {
+                            saveRotation(getCurrentFile(), tmpFile)
+                        } else {
+                            saveFile(tmpFile, bitmap, it as FileOutputStream)
+                        }
+
+                        if (tmpFile.length() > 0 && selectedFile.exists()) {
+                            deleteFile(selectedFile) {}
+                        }
+                        copyFile(tmpFile, selectedFile)
+                        scanPath(selectedFile.absolutePath) {}
+                        deleteFile(tmpFile) {}  // Not necessary?
```

Comment:
guess you can create a "finally" block after the 2 catch blocks and replace 3 "deleteFile(tmpFile)" with 1 in it